test/cmd/diagnostics.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+# This test validates the diagnostics command
+
+# available diagnostics (2015-12-21):
+# AnalyzeLogs ClusterRegistry ClusterRoleBindings ClusterRoles ClusterRouter ConfigContexts DiagnosticPod MasterConfigCheck MasterNode NodeConfigCheck NodeDefinitions UnitStatus
+# Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
+# ConfigContexts runs into some invalid contexts created by other tests, so is omitted.
```

Comment:
Well... that's not entirely true.  test-cmd is about testing functional
scenarios between CLI and server.  So anything between the two can be
deeply tested (as long as its fast).

On Thu, Jan 14, 2016 at 1:12 PM, David Eads <notifications@github.com>
wrote:

> In test/cmd/diagnostics.sh
> <https://github.com/openshift/origin/pull/6397#discussion_r49761876>:
>
> > +
> > +set -o errexit
> > +set -o nounset
> > +set -o pipefail
> > +
> > +OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
> > +source "${OS_ROOT}/hack/util.sh"
> > +source "${OS_ROOT}/hack/cmd_util.sh"
> > +os::log::install_errexit
> > +
> > +# This test validates the diagnostics command
> > +
> > +# available diagnostics (2015-12-21):
> > +# AnalyzeLogs ClusterRegistry ClusterRoleBindings ClusterRoles ClusterRouter ConfigContexts DiagnosticPod MasterConfigCheck MasterNode NodeConfigCheck NodeDefinitions UnitStatus
> > +# Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
> > +# ConfigContexts runs into some invalid contexts created by other tests, so is omitted.
>
> @deads2k <https://github.com/deads2k> @smarterclayton
> <https://github.com/smarterclayton> thoughts on these sorts of
> interactions between tests in test/cmd?
>
> @sosiouxme <https://github.com/sosiouxme> I'd still recommend writing an
> end-to-end test instead, as it would allow you to run tests for your
> diagnostics that require nodes/pods running.
>
> I don't think we need to worry about it too much for test-cmd. test-cmd
> is about seeing if the basic structure of the command works. It's not
> intended to be an exhaustive check of functionality.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6397/files#r49761876>.
>